python_scripts/shellies_discovery.py

```diff
@@ -3124,50 +3124,50 @@ def mqtt_publish(topic, payload, retain, json=False):
-            + str(MAX_TRANSITION)
-            + ')}}{%endif%}}",'
-            '"cmd_off_tpl":"{\\"turn\\":\\"off\\"{%if transition is defined%},\\"transition\\":{{min(transition|multiply(1000),'
-            + str(MAX_TRANSITION)
-            + ')}}{%endif%}}",'
-            '"stat_tpl":"{%if value_json.ison%}on{%else%}off{%endif%}",'
-            '"bri_tpl":"{{value_json.gain|float|multiply(2.55)|round}}",'
-            '"r_tpl":"{{value_json.red}}",'
-            '"g_tpl":"{{value_json.green}}",'
-            '"b_tpl":"{{value_json.blue}}",'
-            '"fx_tpl":"{%if value_json.effect==1%}Meteor Shower{%elif value_json.effect==2%}Gradual Change{%elif value_json.effect==3%}Flash{%else%}Off{%endif%}",'
-            '"uniq_id":"' + unique_id + '",'
-            '"qos":"' + str(qos) + '",'
-            '"dev": {"cns":[["' + KEY_MAC + '","' + format_mac(mac) + '"]],'
-            '"name":"' + device_name + '",'
-            '"mdl":"' + model + '",'
-            '"sw":"' + fw_ver + '",'
-            '"mf":"' + ATTR_MANUFACTURER + '"},'
-            '"~":"' + default_topic + '"}'
-        )
+        payload = {
+            KEY_NAME: light_name,
+            KEY_AVAILABILITY_TOPIC: availability_topic,
+            KEY_PAYLOAD_AVAILABLE: VALUE_TRUE,
+            KEY_PAYLOAD_NOT_AVAILABLE: VALUE_FALSE,
+            KEY_COMMAND_TOPIC: command_topic,
+            KEY_STATE_TOPIC: state_topic,
+            "state_value_template": "{{value.lower()}}",
+            KEY_PAYLOAD_ON: VALUE_ON,
+            KEY_PAYLOAD_OFF: VALUE_OFF,
+            "rgbw_cmd_t": set_topic,
+            "rgbw_cmd_tpl": "{^red^:{{red}},^green^:{{green}},^blue^:{{blue}},^white^:{{white}}}",
+            "rgbw_stat_t": status_topic,
+            "rgbw_val_tpl": "{{value_json.red}},{{value_json.green}},{{value_json.blue}},{{value_json.white}}",
+            "bri_stat_t": status_topic,
+            "bri_val_tpl": "{{value_json.gain|float|multiply(2.55)|round(0)}}",
+            "bri_cmd_t": set_topic,
+            "bri_cmd_tpl": "{^gain^:{{value|float|multiply(0.3922)|round(0)}}}",
+            "fx_cmd_t": set_topic,
+            "fx_cmd_tpl": "{ {%if value==^Off^%}^effect^:0{%elif value==^Meteor Shower^%}^effect^:1{%elif value==^Gradual Change^%}^effect^:2{%elif value==^Flash^%}^effect^:3{%endif%} }",
+            "fx_list": ["Off", "Meteor Shower", "Gradual Change", "Flash"],
+            "fx_stat_t": status_topic,
+            "fx_val_tpl": "{%if value_json.effect==1%}Meteor Shower{%elif value_json.effect==2%}Gradual Change{%elif value_json.effect==3%}Flash{%else%}Off{%endif%}",
+            KEY_UNIQUE_ID: unique_id,
+            KEY_QOS: qos,
+            KEY_DEVICE: device_info,
+            "~": default_topic,
+        }
     else:
         payload = ""
     if dev_id.lower() in ignored:
         payload = ""
 
-    mqtt_publish(config_topic, payload, retain)
+    mqtt_publish(config_topic, payload, retain, json=True)
```

Comment:
@bieniu your code worked, it was just missing this. So the command being sent to the `set` topic was not proper json.